### PR TITLE
V0.13.0

### PR DESCRIPTION
**Pull request by cbouss**
Changes:
- Update version to 0.13.0
- Set use of dpcpp 2022.1 in the feedstock cbc.
- Patch based on upstream changes for compatibility with recent skbuild.
- Patch to use pybind11 from default rather than using git to download it.
- Skip Windows (pending MKL rebuild effort).
- Update dependencies.
- Update build and test scripts.
- Update about section.

Dev URL: https://github.com/IntelPython/dpctl/tree/0.13.0
Upstream recipe: https://github.com/IntelPython/dpctl/blob/0.13.0/conda-recipe/
Additional settings for tests: https://github.com/IntelPython/dpctl/blob/0.13.0/.github/workflows/conda-package.yml#L164

Running tests is disabled through abs.yaml. 
The tests will fail if run on the CI worker, but I tested on a proper x86 server that this works as expected.

### CODE REVIEW

File: abs.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# --no-test because the tests fail when executed from the EC2 instance
+# Tested on an x86 server that this works as expected.
+build_parameters:
+  - "--suppress-variables"
+  - "--error-overlinking"
+  - "--error-overdepending"
+  - "--no-test"
+
+channels: 
+  cbouss: dpcpp
```
Comment (katietz) [inline abs.yaml]: don't miss to clean up channel here on final build

File: recipe/meta.yaml [hunk 2]
```diff
@@ -10,30 +10,44 @@ source:
...
-        - {{ compiler('dpcpp') }}  # [not osx]
+        - {{ compiler('c') }}
+        - {{ compiler('dpcpp') }}
+        - patch     # [not win]
+        - m2-patch  # [win]
     host:
         - setuptools
         - cython
-        - cmake
+        - cmake  >=3.21
```
Comment (katietz) [inline recipe/meta.yaml]: use just one space between package-name and version constrain

File: recipe/meta.yaml [hunk 3]
```diff
@@ -10,30 +10,44 @@ source:
...
+        - pybind11 2.9.2
+        - git
```
Comment (katietz) [inline recipe/meta.yaml]: shouldn't be git, cmake, and other build-tools moved to build-section instead? seeing them in host feels bad

File: recipe/meta.yaml [hunk 4]
```diff
@@ -10,30 +10,44 @@ source:
...
-        - ninja                    # [win]
-        - numpy 1.19
+        - ninja
+        - scikit-build
+        - numpy 1.21
         - wheel
     run:
         - python
-        - {{ pin_compatible('numpy') }}
+        - {{ pin_compatible('numpy', min_pin='x.x', max_pin='x') }}
```
Comment (katietz) [inline recipe/meta.yaml]: use here please instead 'numpy >=1.21,<2'.  We want to get rid of 'pin_compatible'

File: recipe/run_test.bat [hunk 5]
```diff
@@ -1,7 +1,10 @@
...
 
-"%PYTHON%" -c "import dpctl"
+"%PYTHON%" -c "import dpctl; print(dpctl.__version__)"
 if errorlevel 1 exit 1
 
-python -m pytest -q -ra --disable-warnings --pyargs dpctl -vv
+"%PYTHON%" -c "import dpctl; dpctl.lsplatform()"
 if errorlevel 1 exit 1
+
+python -m pytest -q -p no:faulthandler -ra --disable-warnings --pyargs dpctl -vv
```
Comment (katietz) [inline recipe/run_test.bat]: why do we use here suddenly 'build'-sections python?  as above we actually are using the variant from 'host' (%PYTHON% is the host one)

File: recipe/run_test.sh [hunk 6]
```diff
@@ -1,6 +1,10 @@
 #!/bin/bash
 
-set -e
+set -ex
 
-${PYTHON} -c "import dpctl"
-python -m pytest -q -ra --disable-warnings --cov dpctl --cov-report term-missing --pyargs dpctl -vv
```
Comment (katietz) [inline recipe/run_test.sh:6]: why using here system's python?  we should use here ${PYTHON} too, shouldn't we?

**Review by katietz — CHANGES_REQUESTED** [review]

File: recipe/meta.yaml (hunk 3 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml]: `ninja` too -> `requirements/build`

**Comment by skupr-anaconda** [comment]
```
CMake Error at /croot/dpctl_1667572242117/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/python3.9/site-packages/skbuild/resources/cmake/FindCython.cmake:65 (message):

....


`ModuleNotFoundError: No module named 'Cython'`
```

Does it mean that we need `cython` in `build`?

**Comment by cbouss** [comment]
> ```
> CMake Error at /croot/dpctl_1667572242117/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/python3.9/site-packages/skbuild/resources/cmake/FindCython.cmake:65 (message):
> 
> ....
> 
> 
> `ModuleNotFoundError: No module named 'Cython'`
> ```
> 
> Does it mean that we need `cython` in `build`?

It does, I was checking the build locally.

**Review by katietz — APPROVED** [review]
thanks

**Review by varlackc — APPROVED** [review]
LGTM

**Pull request merged**